src/main/java/org/takes/facets/auth/PsBasic.java

```diff
@@ -0,0 +1,175 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.facets.auth;
+
+import java.io.IOException;
+import javax.xml.bind.DatatypeConverter;
+import lombok.EqualsAndHashCode;
+import org.takes.Request;
+import org.takes.Response;
+import org.takes.misc.Opt;
+import org.takes.rq.RqHeaders;
+
+/**
+ * Pass that checks the user according RFC-2617.
+ *
+ * <p>The class is immutable and thread-safe.
+ *
+ * @author Endrigo Antonini (teamed@endrigo.com.br)
+ * @version $Id$
+ * @since 0.20
+ */
+@EqualsAndHashCode(of = { "entry" })
+public final class PsBasic implements Pass {
+
+    /**
+     * Authorization response HTTP head.
+     */
+    private static final String AUTH_HEAD = "Basic";
+
+    /**
+     * Entry to validate user information.
+     */
+    private final transient PsBasic.Entry entry;
+
+    /**
+     * Ctor.
+     * @param basic Entry
+     */
+    public PsBasic(final PsBasic.Entry basic) {
+        this.entry = basic;
+    }
+
+    @Override
+    public Opt<Identity> enter(final Request request) throws IOException {
+        Opt<Identity> identity = new Opt.Empty<Identity>();
+        try {
+            final String decoded = new String(
+                DatatypeConverter.parseBase64Binary(
+                    new RqHeaders.Smart(
+                        new RqHeaders.Base(request)
+                    ).single("authorization").split(AUTH_HEAD)[1]
+                )
+            ).trim();
+            final String user = decoded.split(":")[0];
+            if (this.entry.check(
+                user,
+                decoded.substring(user.length() + 1)
+            ).has()) {
+                identity = new Opt.Single<Identity>(
+                    new Identity.Simple(
+                        String.format("urn:basic:%s", user)
+                    )
+                );
+            }
+        } catch (final IOException ex) {
+            identity = new Opt.Empty<Identity>();
+        }
+        return identity;
```

Comment:
@yegor256 It's not possible to simple return it right now because I have to return the `HTTP 401 status` and `WWW-Authenticate` header.